TextSelection<T>.on*() methods have return type Selection<T> instead of TextSelection<T>, inhibiting chaining

I encountered an `error TS2739: Type 'Selection<Label>' is missing the following properties from type 'TextSelection<Label>': text, align, verticalAlign` when trying to initialize a `TextSelection` using `Scene.createTextSelection()` with the code below. 

```typescript 
import {Scene, SpriteView, TextSelection} from 'megaplot';

interface Label { /* some props */ }

interface Selections {
  labels: TextSelection<Label>;
}

function dummyOnExit(sprite: SpriteView, datum: Label) {}

class SomeClass {
  private readonly scene: Scene;
  private readonly selections: Selections;

  constructor(container: HTMLElement) {
    this.scene = new Scene({container});
    this.selections = {
      // This assignment will throw a TS2739 compiler error
      labels: this.scene.createTextSelection<Label>()
          .align(() => 'center')         // Returns TextSelection<Label>
          .verticalAlign(() => 'bottom') // Returns TextSelection<Label>
          .onExit(dummyOnExit);          // Returns Selection<Label>
    }
  }
}
```

This is caused by the `interface` declarations for the [`Selection<T>`](https://github.com/PAIR-code/megaplot/blob/dev/src/lib/selection-types.d.ts#L45) and [`TextSelection<T>`](https://github.com/PAIR-code/megaplot/blob/dev/src/lib/text-selection-types.d.ts#L31) types. Unlike conventional [JavaScript `class ... extends`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/extends), where you could annotate the returns with `this` and the types would automatically update, [TypeScript `interface ... extends`](https://www.typescriptlang.org/docs/handbook/2/objects.html#extending-types) copies the exact definitions from the parent class without updating any of the types for the child. Thus, since the `interface TextSelection<T> extends Selection<T>` only declares the `text`, `align`, and `verticalAlign` properties, all of the other properties (e.g., `onExit`) retain the typing from `Selection<T>`. 

This will induce a errors in two scenarios:

1. A TS2739 error when assigning the result of `TextSelection<T>.on*()` to a property with type `TextSelection<T>`.
2. A TS2339 error when chaining `.align()` etc. methods with the result of `TextSelection<T>.on*()`.

None of the demos capture this because 1) they [don't use](https://github.com/PAIR-code/megaplot/blob/dev/src/demo/text-demo.ts#L172) method chaining syntax, and 2) [most of them](https://github.com/PAIR-code/megaplot/tree/dev/examples) are written in JavaScript not TypeScript.

--- a/src/lib/text-selection-types.d.ts
+++ b/src/lib/text-selection-types.d.ts
@@ -18,7 +18,7 @@
  * @fileoverview Types relating to text selections.
  */
 
-import {Selection} from './selection-types';
+import {Selection, SelectionCallback} from './selection-types';
 
 export type AlignmentOption = 'left'|'center'|'right';
 
@@ -29,6 +29,56 @@ export type VerticalAlignmentOption = 'top'|'middle'|'bottom';
  * text. Used for showing labels.
  */
 export interface TextSelection<T> extends Selection<T> {
+  /**
+   * Sets the callback to be invoked every time a sprite is bound to its datum.
+   * The bindCallback is invoked before the specific lifecycle phase callback
+   * (init, enter, update and exit).
+   */
+  onBind: (bindCallback: SelectionCallback<T>) => TextSelection<T>;
+
+  /**
+   * Sets the callback to be invoked before a sprite first appears. This is
+   * called after the onBind callback, and any update to the transition time
+   * will be ignored.
+   */
+  onInit: (initCallback: SelectionCallback<T>) => TextSelection<T>;
+
+  /**
+   * Set the callback to be invoked after the sprite first appears. Changes to
+   * the arrival time will be honored. This allows for transitioning properties
+   * as the sprite pops into existence.
+   */
+  onEnter: (enterCallback: SelectionCallback<T>) => TextSelection<T>;
+
+  /**
+   * Set the callback to be invoked after a previously bound sprite is bound
+   * again. This will not run when a sprite first comes into existence, only
+   * when its bound a subsequent time.
+   */
+  onUpdate: (updateCallback: SelectionCallback<T>) => TextSelection<T>;
+
+  /**
+   * Set the callback to be invoked for sprites that are being unbound.
+   */
+  onExit: (exitCallback: SelectionCallback<T>) => TextSelection<T>;
+
+  /**
+   * Bind the provided data points to the selection. This will cause the enter,
+   * update and exit callbacks to be invoked asynchronously. If a key function
+   * is specified, then data points will be bound to Sprites according to the
+   * key string produced.
+   */
+  bind: (data: T[], keyFn?: (datum: T) => string) => TextSelection<T>;
+
+  /**
+   * Clear any previously bound data and Sprites. Previously bound Sprites will
+   * still have their callbacks invoked. This is equivalent to calling bind()
+   * with an empty array, except that it is guaranteed to drop expsting data and
+   * Sprites, whereas calling bind([]) may be interrupted by a later call to
+   * bind().
+   */
+  clear: () => TextSelection<T>;
+
   /**
    * Set the callback for determining the string for the given datum.
    */

